mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1353,33 +1378,17 @@ where
     T: AsRef<UtxoCoinFields> + UtxoCommonOps + MarketCoinOps,
 {
     let decimals = coin.as_ref().decimals;
-    let to = coin
-        .address_from_str(&req.to)
-        .map_to_mm(WithdrawError::InvalidAddress)?;
 
     let conf = &coin.as_ref().conf;
-    let is_p2pkh = to.prefix == conf.pub_addr_prefix && to.t_addr_prefix == conf.pub_t_addr_prefix;
-    let is_p2sh = to.prefix == conf.p2sh_addr_prefix && to.t_addr_prefix == conf.p2sh_t_addr_prefix && conf.segwit;
 
-    let script_pubkey = if is_p2pkh {
-        Builder::build_p2pkh(&to.hash)
-    } else if is_p2sh {
-        Builder::build_p2sh(&to.hash)
-    } else {
-        let error = "Expected either P2PKH or P2SH".to_owned();
-        return MmError::err(WithdrawError::InvalidAddress(error));
-    };
+    let to = address_from_str(coin.as_ref(), &req.to).map_to_mm(WithdrawError::InvalidAddress)?;
```

Comment:
Could you please rename this function to `checked_address_from_str`? Maybe even move it to the `UtxoCommonOps`?

Reply:
yeah, will do that 👍

Reply:
I think moving it to `UtxoCommonOps` is too much, or did you mean renaming `address_from_str` in `UtxoCommonOps` too?

Reply:
I forgot that there is the `address_from_str` method in `UtxoCommonOps`, sorry.
But it seems to me that it's not necessary to rename `UtxoCommonOps::address_from_str`.

There is one more thing I would change:
From `checked_address_from_str(coin.as_ref(), &req.to)` to `coin.address_from_str(&req.to)`,
because some coins like `QtumCoin` may override the `address_from_str` method.

Reply:
Done